File: FBSimulatorControlTests/Tests/Unit/FBSimulatorApplicationDescriptorTests.m
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import <XCTest/XCTest.h>

#import <FBControlCore/FBControlCore.h>

#import <FBSimulatorControl/FBSimulatorControl.h>

#import "FBSimulatorControlFixtures.h"
#import "FBSimulatorControlAssertions.h"

@interface FBSimulatorApplicationDescriptorTests : FBSimulatorControlTestCase

@end

@implementation FBSimulatorApplicationDescriptorTests

- (void)testCreatesSampleApplication
{
  FBBundleDescriptor *application = self.tableSearchApplication;
  XCTAssertEqualObjects(application.identifier, @"com.example.apple-samplecode.TableSearch");
  XCTAssertEqualObjects(application.binary.architectures, ([NSSet setWithArray:@[@"i386", @"x86_64"]]));
}

@end

File: FBSimulatorControlTests/Tests/Unit/FBSimulatorApplicationDescriptorTests.swift
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import FBControlCore
@testable import FBSimulatorControl
import XCTest

final class FBSimulatorApplicationDescriptorTests: FBSimulatorControlTestCase {

  func testCreatesSampleApplication() {
    let application = self.tableSearchApplication()
    XCTAssertEqual(application.identifier, "com.example.apple-samplecode.TableSearch")
    XCTAssertEqual(application.binary!.architectures, Set([FBBinaryArchitecture(rawValue: "i386"), FBBinaryArchitecture(rawValue: "x86_64")]))
  }
}
